kimchi/src/lagrange_basis_evaluations.rs

```diff
@@ -33,10 +100,11 @@ impl<F: FftField> LagrangeBasisEvaluations<F> {
         let stride = p.evals.len() / self.evals[0].len();
         self.evals
             .iter()
-            .map(|evals| {
+            .enumerate()
+            .map(|(j, evals)| {
                 let mut result = F::zero();
                 for (i, e) in evals.iter().enumerate() {
-                    if !p.evals[stride * i].is_zero() {
+                    if !p.evals[stride * j + i].is_zero() {
```

Comment:
:eyes:

Reply:
Oh thanks for pointing out! I left that line after doing some prints, will delete it asap